packages/devtools-reps/src/object-inspector/index.js

```diff
@@ -196,12 +209,29 @@ class ObjectInspector extends Component {
     if (expand === true) {
       const {
         getObjectProperties,
+        getObjectEntries,
         loadObjectProperties,
+        loadObjectEntries,
       } = this.props;
 
       const value = getValue(item);
+      const parent = getParent(item);
+      const parentValue = getValue(parent);
+      const parentActor = parentValue
+        ? parentValue.actor
+        : null;
+
       if (nodeHasProperties(item) && value && !getObjectProperties(value.actor)) {
         loadObjectProperties(value);
+        return;
+      }
+
+      if (
+        nodeIsEntries(item)
+        && !hasAllEntriesInPreview(parent)
+        && !getObjectEntries(parentActor)
+      ) {
+        loadObjectEntries(parentValue);
```

Comment:
We can, but we won't load properties and entries at the same time.
When we expand a map, we load its properties (like any objects/arrays). So the `foo` property will show up there.
This gives us : 
```
▼ Map {1 → 2} 
     foo: "foo"
     size: 1
     ▶︎ <entries>
```

Then we load the entries if the user click on the `<entries>` node.